aws-opentelemetry-distro/src/amazon/opentelemetry/distro/aws_opentelemetry_configurator.py

```diff
@@ -407,6 +408,14 @@ def _customize_span_processors(provider: TracerProvider, resource: Resource) ->
     if _is_lambda_environment():
         provider.add_span_processor(AwsLambdaSpanProcessor())
 
+    # Add session.id baggage attribute to span attributes to support AI Agent use cases
+    # enabling session ID tracking in spans.
+    if is_agent_observability_enabled():
+        def session_id_predicate(baggage_key: str) -> bool:
+            return baggage_key == "session.id"
```

Comment:
Can you add some test cases to cover that only `session.id` gets added from baggage to spans?